official/rnn_ptb/util.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Utility functions that perform various calculations."""
+
+import collections
+
+import tensorflow as tf
+
+EOS = '<eos>'  # End of sequence symbol
+UNK = '<unk>'  # Unknown symbol
+N = 'N'
+DS = '$'
+
+SPECIAL_SYMBOLS = [UNK, N, DS, EOS]
+
+
+def build_vocab_id_dict(data_file):
+  """Create a dictionary mapping all words in the vocabulary to integer IDs."""
+  assert tf.gfile.Exists(data_file), 'File %s does not exist.' % data_file
+  with tf.gfile.Open(data_file, 'r') as f:
+    unique_words = set(f.read().split())
+
+    # Remove special symbols from the set.
+    for symbol in SPECIAL_SYMBOLS:
+      if symbol in unique_words:
+        unique_words.remove(symbol)
```

Comment:
nit: I think this loop is equivalent to `unique_words = unique_words - set(SPECIAL_SYMBOLS)`

Reply:
Nice! I've made this change.